src/factory/src/api.rs

```diff
@@ -43,12 +47,13 @@ impl TokenFactoryCanister {
 
     /// Returns the token, or None if it does not exist.
     #[query]
-    async fn get_token(&self, name: String) -> Option<Principal> {
-        self.state.borrow().factory.get(&name)
+    async fn get_token<'a>(&'a self, name: String) -> Option<Principal> {
```

Comment:
Why is lifetime needed here and below?

Reply:
An implementation detail needed for method exporting because rust cannot infer a lifetime from `self` for `Pin<Box<impl Future<Output = T>> + 'self>>` and will require it to be `'static`.